File: app/src/main/java/com/ardakaplan/myapplication/ui/BaseActivity.java
package com.ardakaplan.myapplication.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.ardakaplan.rdalogger.RDALogger;

/**
 * Created by Arda Kaplan at 4.08.2021 - 13:34
 * <p>
 * ardakaplan101@gmail.com
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onStart() {
        super.onStart();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onRestart() {
        super.onRestart();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onResume() {
        super.onResume();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onPause() {
        super.onPause();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onStop() {
        super.onStop();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        RDALogger.logLifeCycle(getClass().getSimpleName());
    }
}

File: app/src/main/java/com/ardakaplan/testapp/rdalogger/ui/BaseActivity.kt
package com.ardakaplan.testapp.rdalogger.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.ardakaplan.library.rdalogger.RDALogger

/**
 * Created by Arda Kaplan on 21/02/2025 at 14:31.
 *
 * ardakaplan101@gmail.com
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState : Bundle?) {
        super.onCreate(savedInstanceState)

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onStart() {
        super.onStart()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onRestart() {
        super.onRestart()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onResume() {
        super.onResume()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onPause() {
        super.onPause()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onStop() {
        super.onStop()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

    override fun onDestroy() {
        super.onDestroy()

        RDALogger.logLifeCycle(javaClass.simpleName)
    }

}